src/wikitextprocessor/luaexec.py

```diff
@@ -242,24 +242,38 @@ def get_page_content(ctx: "Wtp", title: str) -> Optional[str]:
     return ctx.read_by_title(title.strip())
 
 
-def fetch_language_name(ctx: "Wtp", code: str) -> str:
-    """This function is called from Lua code as part of the mw.language
-    implementation.  This maps a language code to its name."""
-    ret = ctx.LANGUAGES_BY_CODE.get(code)
-    if ret:
-        return ret[0]
-    return ""
+def fetch_language_name(
+    ctx: "Wtp", code: str, inLanguage: Optional[str]
+) -> str:
+    """
+    This function is called from Lua code as part of the mw.language
+    implementation.  This maps a language code to its name.
+    https://www.mediawiki.org/wiki/Extension:Scribunto/Lua_reference_manual#mw.language.fetchLanguageName
+
+    pyicu: https://gitlab.pyicu.org/main/pyicu
+    icu: https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/index.html
+    """
+    from icu import Locale
```

Comment:
Googling it, there doesn't seem to be much of a difference in resource-use or speed whether imports happen at the start of a module or inside a function (because the import is cached anyhow), but we don't do much of this kind of importing inside function elsewhere in the code so it feels better to move this to the start of the file with the others, as `import icu` so that it doesn't become too unclear that `icu.Locale` isn't an import when the import isn't inside the function.

Reply:
Python doesn't have lazy import, import dependencies when using them would decrease startup time. But I add the import inside function mainly because it's external package and it's only used in these two functions. I think other dependencies like `lxml`, `requests` and `pyutil` are also imported in this way.